File: examples/standalone/server-rust/src/procedures/greet.rs
/* examples/standalone/server-rust/src/procedures/greet.rs */

use seam_server::{seam_procedure, SeamError, SeamType};
use serde::{Deserialize, Serialize};

#[derive(Deserialize, SeamType)]
pub struct GreetInput {
  pub name: String,
}

#[derive(Serialize, SeamType)]
pub struct GreetOutput {
  pub message: String,
}

#[seam_procedure]
pub async fn greet(input: GreetInput) -> Result<GreetOutput, SeamError> {
  Ok(GreetOutput { message: format!("Hello, {}!", input.name) })
}

File: examples/standalone/server-bun/src/procedures/greet.ts
/* examples/standalone/server-bun/src/procedures/greet.ts */

import { t } from '@canmi/seam-server'
import type { QueryDef } from '@canmi/seam-server'

export const greet: QueryDef<{ name: string }, { message: string }> = {
	input: t.object({ name: t.string() }),
	output: t.object({ message: t.string() }),
	handler: ({ input }) => {
		return { message: `Hello, ${input.name}!` }
	},
}
